diff --git a/llvm/utils/lit/lit/TestRunner.py b/llvm/utils/lit/lit/TestRunner.py
index 35afa033cf57..4edcfbb3e0e0 100644
--- a/llvm/utils/lit/lit/TestRunner.py
+++ b/llvm/utils/lit/lit/TestRunner.py
@@ -1017,8 +1017,10 @@ def executeScript(test, litConfig, tmpBase, commands, cwd):
             command = litConfig.valgrindArgs + command
 
     try:
+        dredd_env=test.config.environment.copy()
+        dredd_env['DREDD_MUTANT_TRACKING_FILE'] = os.environ['DREDD_MUTANT_TRACKING_FILE']
         out, err, exitCode = lit.util.executeCommand(command, cwd=cwd,
-                                       env=test.config.environment,
+                                       env=dredd_env,
                                        timeout=litConfig.maxIndividualTestTime)
         return (out, err, exitCode, None)
     except lit.util.ExecuteCommandTimeoutException as e:
diff --git a/llvm/utils/lit/lit/main.py b/llvm/utils/lit/lit/main.py
index 41f124a27ad7..d3fdda9d91b7 100755
--- a/llvm/utils/lit/lit/main.py
+++ b/llvm/utils/lit/lit/main.py
@@ -106,7 +106,7 @@ def main(builtin_params={}):
     run_tests(selected_tests, lit_config, opts, len(discovered_tests))
     elapsed = time.time() - start
 
-    record_test_times(selected_tests, lit_config)
+    #record_test_times(selected_tests, lit_config)
 
     if opts.time_tests:
         print_histogram(discovered_tests)
